### ISSUE TITLE
RB35S lua / telemetry bug report

### ISSUE THREAD

**Issue by noamkrief** [FrSkyRC/ETHOS-Feedback-Community#3028]
TDMX to RB35S in both FBUS or SBUS mode.

The lua script that controls the RB35S is acting buggy.
Options go from "live" to grayed out at random times.
Repeated by radio calling "telemetry lost" "telemetry restored" etc.

It makes it impossible to maneuver in the RB35S lua setup pages.

If I set my channel count 1-22 / 1-23 / 1-24 I get this behavior.

If I set my channel count to 21 or less, this buggy behavior disappears.

Hope you can replicate on your end.

**Comment by bsongis-frsky** [comment]
It seems to me not an Ethos bug, I will forward it to the RF team

**Comment by SmartTommy** [comment]
> TDMX to RB35S in both FBUS or SBUS mode.
> 
> The lua script that controls the RB35S is acting buggy. Options go from "live" to grayed out at random times. Repeated by radio calling "telemetry lost" "telemetry restored" etc.
> 
> It makes it impossible to maneuver in the RB35S lua setup pages.
> 
> If I set my channel count 1-22 / 1-23 / 1-24 I get this behavior.
> 
> If I set my channel count to 21 or less, this buggy behavior disappears.
> 
> Hope you can replicate on your end.

I just tried to repeat the problem you described, but it didn't happen. Can you give us a video?

**Comment by mawzthefinn** [comment]
@bsongis-frsky There's an ETHOS bug here, X20 Pro should not be able to set a channel range of 1-22 or 1-21, only 1-8, 1-16 or 1-24 when in TD mode. I've confirmed my X20 Pro can set an invalid channel range, which we fixed in the older radios a while back

**Comment by noamkrief** [comment]
@SmartTommy 
@bsongis-frsky 
Here is the video showing the problem.
Thank you so much for investigating it :)
https://youtu.be/wwx91WF8mVM

**Comment by noamkrief** [comment]
@bsongis-frsky 
Yes - if you watch the video I posted above, you'll see the bug occurring when I use 1-24

**Comment by mawzthefinn** [comment]
#3005 looks like the same issue, works once channel range is set to 1-21

**Comment by ACCURA68** [comment]
Hello everyone, I had exactly the same error with my x20 pro and an rb30 and assigning channels 1 to 21 in the RF system and in the tdmx sbus 24, both primary and secondary, everything works correctly, I just had to deactivate the telemetry of the secondary TDmx because it constantly gave sensor losses

**Comment by noamkrief** [comment]
@mawzthefinn 
@ACCURA68 

So let see if I understand, because I'm brand new to frsky.
We should be able to only set "channel count" to 1-8, 1-16, or 1-24
But there is a bug so the software lets us select WHATEVER we want.

And for some reason 1-21 is a magic number which makes the "bug" go away.

Once they force us via software to only select 1-8,1-16 or 1-24 we will be stuck with 1-16 rather than 1-21

**Comment by mawzthefinn** [comment]
The only channel ranges that are properly valid for ACCESS< TANDEM or TWIN are 1-8, 1-16 and 1-24 (which is 1, 2 or 3 frames at 8ch per frame), there is a bug with the settings which is causing issues with channel ranges from 1-22 through 1-24 and the ability to set incorrect channel ranges.

The fix should address both the ability to set incorrect channel ranges and the bug which is currently occurring when 1-22, 1-23 or 1-24 is set. 

Once fixed, 1-24 should work correctly.

**Comment by mawzthefinn** [comment]
NB - 1.4.14 does not fix the underlying issue on this one, right now it's looking like an RF module issue. It has been referred to the correct engineers. 

I've tested the X20 Pro vs the X18 on my RB35S Testbed aircraft (2 TDMX's and an RB35S), and confirmed that the X20PRO will drop link when configuring if the channel range is 1-24, but the X18 does not do the same thing. 

If you are using the ch1-21 workaround, hold off in 1.4.14 for now.

**Comment by noamkrief** [comment]
@mawzthefinn Just to be clear, I should NOT update my X20Pro to 1.4.14 because i'll lose the ability to set 1-21?
Will I be notified here when there is an update that will fix this issue for me? Perhaps 1.4.15?

**Comment by mawzthefinn** [comment]
I will update this issue, more likely to be TD-ISRM 1.0.12 or 1.0.13 that fixes it

**Comment by mawzthefinn** [comment]
Testing with TD ISRM PRO 1.0.12 today.

It is NOT compatible with TDMX firmware 1.0.10 or earlier (don't fly this combo, RSSI is bad), it does work well with TDMX 1.0.11

In my testing, it is working correctly with a channel range of 1-24 in that combo.

**Comment by ACCURA68** [comment]
Thanks a lot Adam, take note.

Obtener TypeApp para Android<http://www.typeapp.com/r>
En 20 nov. 2023, en 18:44, mawzthefinn ***@***.******@***.***>> escribió:

Testing with TD ISRM PRO 1.0.12 today.

It is NOT compatible with TDMX firmware 1.0.10 or earlier (don't fly this combo, RSSI is bad), it does work well with TDMX 1.0.11

In my testing, it is working correctly with a channel range of 1-24 in that combo.

**Comment by noamkrief** [comment]
Two issues: (updated my x20pro to TD Pro 1.1.12)
1) When updating TDMX (HW 1.6.0) to FW v1.0.11, after update via wire the TDMX becomes inop. No LED.
To get the TDMX back to life, must downgrade back to v1.0.10

Again - TDMW HW 1.6.0 cannot take the 1.0.11 update because it died (no LED)

2) Updated TDR6 FW to v1.0.11 and connected to RB35S via FBUS. Lua script still crashes, and complete loss to RB35S telemetry. Same symptom as original issue. (SBUS mode seems more reliable. Crashes less often)

**Comment by SmartTommy** [comment]
> Two issues: (updated my x20pro to TD Pro 1.1.12)
> 
> 1. When updating TDMX (HW 1.6.0) to FW v1.0.11, after update via wire the TDMX becomes inop. No LED.
>    To get the TDMX back to life, must downgrade back to v1.0.10
> 
> Again - TDMW HW 1.6.0 cannot take the 1.0.11 update because it died (no LED)
> 
> 2. Updated TDR6 FW to v1.0.11 and connected to RB35S via FBUS. Lua script still crashes, and complete loss to RB35S telemetry. Same symptom as original issue. (SBUS mode seems more reliable. Crashes less often)

1. thank you for your test, new firmware will support the HW1.6.
2. please make sure chanel range is 1~8/1-16/1~24, never use the other channel range.

**Comment by noamkrief** [comment]
@SmartTommy Yes on 1-8,1-16,1-24 actually with a recent X20pro update, I lose the capacity to do anything else like 1-17. It's impossible now :)

If you would like, I can make a video of TDR6 with latest firmware failing on the RB35S lua.
when the LUA stops responding, all communications to the RB35S fail. So "Ch1 current" etc all turn red. Reboot of RB35S brings it all back.

**Comment by SmartTommy** [comment]
[TDMX_1.0.11_20231121.zip](https://github.com/FrSkyRC/ETHOS-Feedback-Community/files/13422374/TDMX_1.0.11_20231121.zip)

**Comment by noamkrief** [comment]
@SmartTommy 
This new firmware keeps the TDMX alive :) that's good news.
But the underlying issue is even worse now.
I can get telemetry from RB35S  (ch1,2,3.... currents) in 1-8 1-16 1-24 but in NONE of these combinations does the LUA settings work.
Again - LUA settings are all grayed out.

I tried in FBUS and SPORT mode. Behavior is the same.


![x20pro](https://github.com/FrSkyRC/ETHOS-Feedback-Community/assets/149995415/7d1bbe41-3fea-4006-89c2-df9357a3a89b)

**Comment by noamkrief** [comment]
@SmartTommy 
@mawzthefinn 

Confirming is all working perfect now! Lua very responsive even in 1-24

THANK YOU!!!!!!!!!!!!!!!!!

**Comment by noamkrief** [comment]
@SmartTommy 
@mawzthefinn 

Somehow the TDR6 is working perfect now too even in 1-24 both FBUS and SPORT

X20pro ISRM-pro 1.1.12
TDR6 HW 1.1.0 FW 1.0.11

This is all WORKING now!
Guys - amazing work and super quick! What a great company and people!
You may close this ticket :)
Noam

**Comment by bsongis-frsky** [comment]
Thanks for the feedback!
